src/sampling.py

```diff
@@ -22,6 +22,51 @@ def mnist_iid(dataset, num_users):
         all_idxs = list(set(all_idxs) - dict_users[i])
     return dict_users
 
+def split_dirichlet(dataset, num_users, is_cfar, beta=0.5):
+    """
+    Sample non-I.I.D client data from an arbitary dataset.
+    Samples it based on this paper: 10.48550/ARXIV.1905.12022
+    :param dataset: The dataset
+    :param num_users: The number of clients
+    :param is_cfar: Whether the dataset is cfar (bool). This is a stupid solution but is necessary since for some reason,
+    their parameter name is different.
+    :return: dict mapping client id to idxs for training
+    """
+    idxs = np.arange(len(dataset))
+    labels = np.array(dataset.targets) if is_cfar else dataset.train_labels.numpy()
+    uniq_labels = np.unique(labels)
+    dict_users = {i: np.array([]) for i in range(num_users)}
+    idxs_labels = np.vstack((idxs, labels))
+    idxs_labels = idxs_labels[:, idxs_labels[1, :].argsort()]
+    idxs_labels = idxs_labels.transpose()
```

Comment:
Why do need to transpose  the idxs_labels?

Reply:
It's vertically stacked beforeand ig it looks nicer when its a table (with idxs and labels as seperate columns?)

Reply:
Exactly, it's used for the next section when filtering through the array. Once filtering is done, it is transposed back anyways.